src/onnx/parse_resize.cpp

```diff
@@ -33,49 +33,28 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace onnx {
 
-static std::vector<int>
-calc_neighbor_points(const std::vector<std::vector<std::vector<std::size_t>>>& vvv_ind,
-                     int i_dim,
-                     std::vector<std::vector<std::size_t>> vec_dims,
-                     const shape& in_s)
+static void calc_neighbor_points(const std::vector<std::vector<std::vector<std::size_t>>>& vvv_ind,
+                                 const std::size_t& n_dims,
+                                 const std::size_t& out_elements,
+                                 const shape& in_s,
+                                 std::vector<int>& vec_ind)
 {
-    if(i_dim == vvv_ind.size())
+    for(std::size_t start = 0; start < (std::size_t{1} << n_dims); start++)
     {
-        std::vector<int> vec_ind(vec_dims.size());
-        std::transform(vec_dims.begin(), vec_dims.end(), vec_ind.begin(), [&](auto idx) {
-            return static_cast<int>(in_s.index(idx));
-        });
-        return vec_ind;
-    }
-
-    const auto& vv_lo = vvv_ind[i_dim][0];
-    std::vector<std::vector<std::size_t>> vec_dims1;
-    for(std::size_t start = 0; start < vec_dims.size(); start += vv_lo.size())
-    {
-        std::transform(vv_lo.begin(),
-                       vv_lo.end(),
-                       vec_dims.begin() + start,
-                       std::back_inserter(vec_dims1),
-                       [](auto i, auto dim) {
-                           dim.push_back(i);
-                           return dim;
-                       });
-    }
-
-    const auto& vv_hi = vvv_ind[i_dim][1];
-    for(std::size_t start = 0; start < vec_dims.size(); start += vv_hi.size())
-    {
-        std::transform(vv_hi.begin(),
-                       vv_hi.end(),
-                       vec_dims.begin() + start,
-                       std::back_inserter(vec_dims1),
-                       [](auto i, auto dim) {
-                           dim.push_back(i);
-                           return dim;
-                       });
+        std::vector<std::size_t> idx(n_dims);
+        for(std::size_t idx_e = 0; idx_e < out_elements; idx_e++)
```

Comment:
Rename idx_e to something else as this makes it harder to follow when you have idx in the above loop and idx used right after with idx_e as one of the indicies in the access. At glance its easy to miss because of the similarity. 

Something like `elements_idx` would be clearer

Reply:
Good point! Yes let me rename the variable.